vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -89,4 +99,32 @@ public ISequence checkIfSequenceExists(String moduleId, String sequenceId) {
     public Page<IModule> findByNameOrDescription(Pageable requestedPage,String searchText) {
         return moduleRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage,searchText,searchText);
     }
+    
+    @Override
+    public void deleteModule(String moduleId) {
+        List<ModuleLink> moduleLinksToPersist = new ArrayList<ModuleLink>();
+        if(moduleId != null) {
+            List<IModuleLink> moduleLinks = moduleLinkRepo.findModuleLinkByModuleId(moduleId);
+            for(IModuleLink moduleLink: moduleLinks) {
+                moduleLink.setModule(null);
+                moduleLinksToPersist.add((ModuleLink)moduleLink);
+            }
+            moduleLinkRepo.saveAll(moduleLinksToPersist);
+            //delete all slides
+            Optional<Module> moduleOptional = moduleRepo.findById(moduleId);
+            if(moduleOptional.isPresent()) {
+                Module module = moduleOptional.get();
+                List<ISlide> slides = module.getSlides();
+                for(ISlide slide: slides) {
+                    slideManager.deleteSlideById(slide.getId(), moduleId);
+                }
+                List<ISequence> sequences = getModuleSequences(moduleId);
+                for(ISequence sequence:sequences) {
+                    sequenceRepo.deleteById(sequence.getId());
+                }
+            }
+            moduleRepo.deleteById(moduleId);
+        }
+        return;
+    }
```

Comment:
you want to make sure that all of the above changes are being rolled back if an error happens along the way (we don't want half deleted modules). Either annotate this method or the whole class with (which will rollback the transaction on any exception).
```
@Transactional(rollbackFor = { Exception.class })
```

Reply:
done